lib/fluent/plugin/buffer.rb

```diff
@@ -614,4 +614,7 @@ def write_step_by_step(metadata, data, format, splits_count, &block)
       end
     end
   end
+
+  # Some input plugins refer BufferQueueLimitError for throttling
+  BufferQueueLimitError = ::Fluent::Plugin::Buffer::BufferOverflowError
```

Comment:
Define this constant should be under `lib/fluent/compat` because it's only for compat plugins.